ProcessLib/ComponentTransport/ComponentTransportProcess.cpp

```diff
@@ -174,6 +174,32 @@ void ComponentTransportProcess::
         _local_assemblers, pv.getActiveElementIDs(), _coupled_solutions);
 }
 
+void ComponentTransportProcess::extrapolateIntegrationPointValuesToNodes(
+    const double t, std::vector<GlobalVector*> const& int_pt_x,
+    std::vector<GlobalVector*>& x)
+{
+    auto& extrapolator = getExtrapolator();
+    auto const extrapolatables =
+        NumLib::makeExtrapolatable(_local_assemblers,
+                                   &ComponentTransportLocalAssemblerInterface::
+                                       getInterpolatedLocalSolution);
+
+    for (unsigned transport_process_id = 0;
+         transport_process_id < int_pt_x.size();
+         ++transport_process_id)
+    {
+        auto const& pv = _process_variables[transport_process_id][0].get();
+        auto const& int_pt_C = int_pt_x[transport_process_id];
+
+        extrapolator.extrapolate(pv.getNumberOfComponents(), extrapolatables, t,
+                                 {int_pt_C},
+                                 {_local_to_global_index_map.get()});
+
+        auto const& nodal_values = extrapolator.getNodalValues();
+        *x[transport_process_id + 1] = nodal_values;
```

Comment:
Since x is a distributed vector under MPI, here you have to use LinAlg::copy(...), which is declared  in MathLib/LinAlg/LinAlg.h. With that, the compilation error under MPI should go.

Reply:
Let me have a try. Thanks, Wenqing!

Reply:
it works out.